tokio-timer/src/timer/handle.rs

```diff
@@ -131,9 +132,15 @@ impl Handle {
 
     /// Create a `Delay` driven by this handle's associated `Timer`.
     pub fn delay(&self, deadline: Instant) -> Delay {
+        self.delay_timeout(deadline, Duration::from_secs(0))
+    }
+
+    fn delay_timeout(&self, deadline: Instant, duration: Duration) -> Delay {
         match self.inner {
-            Some(ref handle_priv) => Delay::new_with_handle(deadline, handle_priv.clone()),
-            None => Delay::new(deadline),
+            Some(ref handle_priv) => {
+                Delay::new_with_handle(deadline, Duration::from_secs(0), handle_priv.clone())
```

Comment:
Should this pass the above duration, instead of creating a 0?

Reply:
The second duration there is the magic reset interval that's used by `Timeout<Stream>`. `Delay`'s normal public constructor sets it to 0 as well.